libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -321,6 +339,13 @@ void AP_GPS::init(const AP_SerialManager& serial_manager)
 {
     primary_instance = 0;
 
+    // reset the GPS ID if not UAVCAN
+    for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+        if (_type[i] != GPS_TYPE_UAVCAN) {
+            set_and_save_devid(i, 0);
```

Comment:
Don't set and save every cycle, this wheres out flash on flash based boards, and generally the user hasn't gained much by this whole loop.